src/teleop/gui/src/components/ArmControls.vue

```diff
@@ -1,99 +1,184 @@
+      joystick_pub: null,
+      joystickservo_pub: null,
+      jointlock_pub: null,
+      joints_array: [false, false, false, false, false, false]
+    };
   },
 
-  beforeDestroy: function () {
-    window.clearInterval(interval)
+  computed: {
+    ...mapGetters("controls", {
+      controlMode: "controlMode"
+    })
   },
 
+  beforeUnmount: function () {
+    window.clearInterval(interval);
+  },
 
   created: function () {
     this.joystick_pub = new ROSLIB.Topic({
-        ros : this.$ros,
-        name : '/xbox/ra_control',
-        messageType : 'sensor_msgs/Joy'
-    })
-    const updateRate = 0.1
+      ros: this.$ros,
+      name: "/xbox/ra_open_loop",
+      messageType: "sensor_msgs/Joy"
+    });
+    this.joystickservo_pub = new ROSLIB.Topic({
+      ros: this.$ros,
+      name: "/xbox/ra_servo",
+      messageType: "sensor_msgs/Joy"
+    });
+    this.jointlock_pub = new ROSLIB.Topic({
+      ros: this.$ros,
+      name: "/joint_lock",
+      messageType: "mrover/JointLock"
+    });
+    const jointData = {
+      //publishes array of all falses when refreshing the page
+      joints: this.joints_array
+    };
+    var jointlockMsg = new ROSLIB.Message(jointData);
+    this.jointlock_pub.publish(jointlockMsg);
+
+    const updateRate = 0.1;
     interval = window.setInterval(() => {
-      if(this.arm_enabled){
-        const gamepads = navigator.getGamepads()
+      if (this.arm_enabled) {
+        const gamepads = navigator.getGamepads();
         for (let i = 0; i < 4; i++) {
-          const gamepad = gamepads[i]
+          const gamepad = gamepads[i];
           if (gamepad) {
-            if (gamepad.id.includes('Microsoft') || gamepad.id.includes('Xbox')) {
-              let buttons = gamepad.buttons.map((button) =>{
-                  return button.value
-                })
-  
-  
+            if (
+              gamepad.id.includes("Microsoft") ||
+              gamepad.id.includes("Xbox")
+            ) {
+              let buttons = gamepad.buttons.map((button) => {
+                return button.value;
+              });
+
               const joystickData = {
-                  axes: gamepad.axes,
-                  buttons: buttons
+                axes: gamepad.axes,
+                buttons: buttons
+              };
+              var joystickMsg = new ROSLIB.Message(joystickData);
+              if (this.servo === true) {
+                this.joystickservo_pub.publish(joystickMsg);
+              } else {
+                this.joystick_pub.publish(joystickMsg);
               }
-              var joystickMsg = new ROSLIB.Message(joystickData)
-              this.joystick_pub.publish(joystickMsg)
             }
           }
         }
       }
-    }, updateRate*1000)
+    }, updateRate * 1000);
   },
 
-    methods: {
-        updateArmEnabled: function (enabled){
-            this.arm_enabled = enabled
-        }
+  methods: {
+    updateArmEnabled: function (enabled) {
+      this.arm_enabled = enabled;
```

Comment:
Probably makes sense to send a zero command to stop motion of the arm immediately if this gets set to false

Reply:
I still think this maybe needs to get done... Would we not want to send a zero command?